src/shared/ClineBanner.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Banner message types for Cline extension
+ */
+
+export type BannerSeverity = "info" | "success" | "warning"
+export type BannerPlacement = "top" | "bottom"
+
+export interface Banner {
+	id: string
+	titleMd: string
+	bodyMd: string
+	severity: BannerSeverity
+	placement: BannerPlacement
+	rulesJson: string
+	activeFrom?: string
+	activeTo?: string
+}
+
+export interface BannersResponse {
+	data: {
+		banners: Banner[]
+	}
+}
+
+/**
+ * Audience targeting options
+ */
+export type BannerAudience = "all" | "team admin only" | "team members" | "personal"
+
+/**
+ * Rules that can be evaluated for banner targeting
+ */
+export interface BannerRules {
```

Comment:
The current implementation goes like this: 
if there is specified attributes e.g. ide, auth, providers, audience, features, it will only show banners to users whose attributes match the specified values. 
And if those attributes are not specified for the banner, it will show banner to ALL users.